### DISCUSSION TITLE
Guéant Lambda Calculation - Why Divide Depth by 0.5?

### DISCUSSION THREAD

**Discussion by EssamEmad** [nkaz001/hftbacktest#46] [Q&A] [answered]
## TL;DR

When estimating the lambda to calibrate A and K, you are dviding the depth by half and taking 70 lambda timepoints. Why?

## Context

First off, thanks a ton for both the OSS backtesting tool and for open-sourcing some examples.
I was trying to understand how you estimate lambda and calibrate the parameters. In [this](https://quant.stackexchange.com/questions/36073/how-does-one-calibrate-lambda-in-a-avellaneda-stoikov-market-making-problem) article, it says "check the time when the price reaches ...". In the example you have, in `measure_trading_intensity`, you seem to:

1. Get the counts of trades with delta ticks away from the mid-price (depth)
2. Divide it by 0.5 and round up
3. Scalar divide lambda by 600
4. Take the first 70 datapoints of lambda
5. Estimate A and K using linear regression with lambda calculated from 4

## Questions
Other than step 1 and 5, I have a few questions:
1. Why do you divide by 0.5?
2. The article says check the time when the price reaches the depth. I really don't understand this statement and how it maps to working out mean # of events per second which you are doing. I get that the latter is assuming a poisson distribution, but we are not looking at any time delta right?
3. Why do you scalar divide by 600? Shouldn't you take the average over time buckets to shrink the window?
4. Why do you take the first 70 datapoints?

Thanks!

**Comment by nkaz001** [comment, marked as answer]
Q1 & Q2:
When we measure order arrival depth from the mid price, the closest possible tick to the mid price is 0.5.
For instance, if the tick size is 1:

* Ask: 10, Bid: 9, Mid: 9.5. 
If a buy trade occurs at 10, the order arrival depth is 0.5.

* Ask: 11, Bid: 9, Mid: 10.
If a sell trade occurs at 9, the order arrival depth is 1.

To count the order arrival for a given time using index operator, I convert 0.5 to 0, 1 to 1, 1.5 to 2, and so forth.

Considering the order fills based on the queue position is intricate when measuring the order arrival depth. So, it's assumed that order fills occur only when the trade passes through a particular price.

For instance, if a buy trade occurs at 12, ask orders below 12 (e.g., 10, 11) are deemed filled. Thus, if your order is at 10, the order arrival count increases by one due to this trade.

This is what `measure_trading_intensity` method does.

Q3:
The order arrival count is totaled for a 10-minute window, meaning it's per 10-minute. To convert it to per second, it is divided by 600(60 * 10 sec).

Q4:
As illustrated in the example, the exponential function isn't suitable for the entire range. Therefore, you need to select an appropriate range to fit your function.
Since our quote would be within 70 ticks (if it's too distant from the mid, it's unlikely our order will be filled), I arbitrarily opt for 70 ticks.
